### PR TITLE
refactoring and clean up

### PR DESCRIPTION
**Pull request by aldoxtor**

### CODE REVIEW

File: rwengine/src/objects/CharacterObject.cpp [hunk 1]
```diff
@@ -480,10 +480,7 @@ float CharacterObject::getJumpSpeed() const {
 }
 
 bool CharacterObject::isOnGround() const {
-    if (physCharacter) {
-        return physCharacter->onGround();
-    }
-    return true;
+    return physCharacter == nullptr || physCharacter->onGround();
```
Comment (JayFoxRox) [inline rwengine/src/objects/CharacterObject.cpp]: This is less readable

File: rwengine/src/engine/Animator.cpp [hunk 2]
```diff
@@ -51,9 +51,6 @@ void Animator::tick(float dt) {
             BoneTransform xform;
             if (b.first->type == AnimationBone::R00) {
                 xform.rotation = kf.rotation;
-            } else if (b.first->type == AnimationBone::RT0) {
-                xform.rotation = kf.rotation;
-                xform.translation = kf.position;
```
Comment (JayFoxRox) [inline rwengine/src/engine/Animator.cpp]: Why was this removed? This is probably a supported version the fallback below might just be a fallback / non-defined behaviour which is still handled

File: rwengine/src/engine/Animator.cpp [hunk 3]
```diff
@@ -89,10 +86,8 @@ void Animator::tick(float dt) {
 
 bool Animator::isCompleted(unsigned int slot) const {
     if (slot < animations.size()) {
-        return animations[slot].animation
-                   ? animations[slot].time >=
-                         animations[slot].animation->duration
-                   : true;
+        return animations[slot].animation == nullptr || animations[slot].time >=
+                                                        animations[slot].animation->duration;
```
Comment (JayFoxRox) [inline rwengine/src/engine/Animator.cpp]: This is less readable imo. Should at least contain braces or made into variables.
I think an early out would also be nice:

```
if (animations[slot].animation == nullptr) {
  return true;
}
return animations[slot].time >= animations[slot].animation->duration;
```

File: rwengine/src/engine/GameWorld.cpp [hunk 4]
```diff
@@ -737,10 +737,7 @@ void GameWorld::clearCutscene() {
 bool GameWorld::isCutsceneDone() {
     if (state->currentCutscene) {
         float time = getGameTime() - state->cutsceneStartTime;
-        if (state->skipCutscene) {
-            return true;
-        }
-        return time > state->currentCutscene->tracks.duration;
+        return state->skipCutscene ? true : time > state->currentCutscene->tracks.duration;
```
Comment (JayFoxRox) [inline rwengine/src/engine/GameWorld.cpp]: Old code was more readable.
We are not trying to win a "shortest code" competition. More explicit is probably better

File: rwengine/src/objects/CharacterObject.cpp [hunk 5]
```diff
@@ -497,7 +494,7 @@ void CharacterObject::setJumpSpeed(float speed) {
 
 void CharacterObject::resetToAINode() {
     auto nodes = engine->aigraph.nodes;
-    bool vehicleNode = !!getCurrentVehicle();
+    bool vehicleNode = getCurrentVehicle() != nullptr;
```
Comment (JayFoxRox) [inline rwengine/src/objects/CharacterObject.cpp]: imo such comparisons in assignments should always be in braces. What does our style guide say @danhedron ?

File: rwengine/src/objects/VehicleObject.cpp [hunk 6]
```diff
@@ -351,7 +351,7 @@ void VehicleObject::tickPhysics(float dt) {
                     if (_lastHeight >= wH) {
                         // we are for real, underwater
                         inWater = true;
-                    } else if (inWater == false) {
+                    } else if (inWater == false) { // This condition is unnecessary
```
Comment (JayFoxRox) [inline rwengine/src/objects/VehicleObject.cpp]: True. Remove it altogether or refactor this code branch?

File: rwgame/states/BenchmarkState.cpp [hunk 7]
```diff
@@ -4,7 +4,7 @@
 #include "RWGame.hpp"
 
 BenchmarkState::BenchmarkState(RWGame* game, const std::string& benchfile)
-    : State(game), benchfile(benchfile), benchmarkTime(0.f), frameCounter(0) {
+    : State(game), benchfile(benchfile), benchmarkTime(0.f), duration(0.f), frameCounter(0) {
```
Comment (JayFoxRox) [inline rwgame/states/BenchmarkState.cpp]: imo we should move more of these constants into the member declaration.
But this is probably fine too.

File: rwviewer/ViewerWidget.cpp [hunk 8]
```diff
@@ -163,7 +165,11 @@ void ViewerWidget::drawFrameWidget(ModelFrame* f, const glm::mat4& m) {
         glLineWidth(1.f);
     }
     dp.textures = {whiteTex};
-    renderer->getRenderer()->drawArrays(thisM, _frameWidgetDraw, dp);
+    
+    RW_CHECK(renderer != nullptr, "GameRenderer is null");
+    if(renderer != nullptr) {
+        renderer->getRenderer()->drawArrays(thisM, _frameWidgetDraw, dp);
+    }
```
Comment (JayFoxRox) [inline rwviewer/ViewerWidget.cpp]: This should probably just error if the GameRenderer is null.

(I think the viewer tools is dead anyway :P )

**Comment by JayFoxRox** [comment]
Please add a PR description. I also think this should not be 3 seperate commits unless there is a pattern / structure in the commit order I'm not seeing.

File: rwengine/src/engine/Animator.cpp (hunk 2 above)
Comment (aldoxtor) [inline rwengine/src/engine/Animator.cpp]: I removed 'else if' because it is doing the same thing like 'else'.

File: rwengine/src/objects/VehicleObject.cpp (hunk 6 above)
Comment (aldoxtor) [inline rwengine/src/objects/VehicleObject.cpp]: this condition should doing something or it is useless?

File: rwviewer/ViewerWidget.cpp (hunk 8 above)
Comment (aldoxtor) [inline rwviewer/ViewerWidget.cpp]: if something is dead it shouldn't be in repo.

File: rwengine/src/objects/VehicleObject.cpp (hunk 6 above)
Comment (JayFoxRox) [inline rwengine/src/objects/VehicleObject.cpp]: The code itself is pretty broken and does not what the comment suggests. The check for line 350 is seemingly never done. We also had issue #214 which is probably affected by this.
So it probably needs a full refactor.
However, seeing how this is mostly a small cleanup, I'd suggest linking this "weird" code from #214 and not worrying about it now (e.g. remove this part from the PR).

**Comment by aldoxtor** [comment]
I correct this PR.

File: rwviewer/ViewerWidget.cpp (hunk 8 above)
Comment (JayFoxRox) [inline rwviewer/ViewerWidget.cpp]: agreed, but that's a seperate issue.
(We are currently transitioning renderers, first internally and then to librw by aap, that broke a ton of stuff)

**Comment by JayFoxRox** [comment]
Sorry for taking so long with the review.

LGTM

File: rwviewer/ViewerWidget.cpp (hunk 8 above)
Comment (danhedron) [inline rwviewer/ViewerWidget.cpp]: The viewer should still be working.

**Comment by danhedron** [comment]
Thanks for tidying up, LGTM.
